src/aura/AuraAvatarMultiToken.sol

```diff
@@ -514,34 +496,48 @@ contract AuraAvatarTwoToken is BaseAvatar, PausableUpgradeable, AuraAvatarUtils,
     /// @param _performData ABI-encoded reference price for AURA in USD (in 8 decimal precision) to compare with TWAP price.
     ///                     The first 4 bytes of the encoding are ignored and the rest is decoded into a uint256.
     function performUpkeep(bytes calldata _performData) external override onlyKeeper whenNotPaused {
-        uint256 auraPriceInUsd = abi.decode(_performData[4:], (uint256));
-        _processRewardsKeeper(auraPriceInUsd);
+        _processRewardsKeeper();
     }
 
     /// @notice A function to process pending BAL and AURA rewards at regular intervals. Can only be called by the
     ///         keeper when the contract is not paused.
-    /// @param _auraPriceInUsd A reference price for AURA in USD (in 8 decimal precision) to compare with TWAP price.
-    function processRewardsKeeper(uint256 _auraPriceInUsd) external onlyKeeper whenNotPaused {
-        _processRewardsKeeper(_auraPriceInUsd);
+    function processRewardsKeeper() external onlyKeeper whenNotPaused {
+        _processRewardsKeeper();
     }
 
     ////////////////////////////////////////////////////////////////////////////
     // PUBLIC VIEW
     ////////////////////////////////////////////////////////////////////////////
 
     /// @notice The total amounts of both BPT tokens that the avatar is handling.
-    function totalAssets() external view returns (uint256 asset1Amounts_, uint256 asset2Amounts_) {
-        asset1Amounts_ = baseRewardPool1.balanceOf(address(this));
-        asset2Amounts_ = baseRewardPool2.balanceOf(address(this));
+    function totalAssets() external view returns (uint256[] memory) {
+        uint256 length = baseRewardPools.length();
+        uint256[] memory assetAmounts = new uint256[](length);
+
+        for (uint256 i = 0; i < length;) {
+            assetAmounts[i] = IBaseRewardPool(baseRewardPools.at(i)).balanceOf(address(this));
+            unchecked {
+                ++i;
+            }
```

Comment:
Nice optimization, but we should keep this conssitent over all the for loops in the contract

Reply:
added the opti across the board in 1bf3520289ab8590692162543359a3039d54602b